pvlib/data/gold/sdm.py

```diff
@@ -0,0 +1,847 @@
+        pvlib.pvsystem.retrieve_sam('cecmod').SunPower_SPR_E20_327)
+    device.name = 'Bishop_{}_{}_{}'.format(N_s, cell_area_cm2, ideality_factor)
+    device.BIPV = 'N'
+    device.Date = '6/20/1988'
+    device.T_NOCT = None
+    device.A_c = None
+    device.N_s = N_s
+    device.I_sc_ref = None
+    device.V_oc_ref = None
+    device.I_mp_ref = None
+    device.V_mp_ref = None
+    device.beta_oc = None
+    device.a_ref = N_s * ideality_factor * boltzmann_J_per_K * T_stc_K / \
+        elementary_charge_C  # Volt
+    device.I_L_ref = cell_area_cm2 * 30.e-3  # Amp
+    device.I_o_ref = cell_area_cm2 * 5500.e-12  # Amp
+    device.R_s = N_s * cell_area_cm2 * 1.33  # Ohm
+    device.R_sh_ref = cell_area_cm2 * 750. / N_s  # Ohm
+    device.Adjust = None
+    device.gamma_r = None
+    device.Version = None
+    device.PTC = None
+
+    return device
+
+
+def bootstrap_si_cell_large():
+    """
+    Create a single-large-area-cell c-Si device based on a SunPower module.
+    """
+
+    # Use a large single cell from an existing mono-Si module
+    device = copy.deepcopy(
+        pvlib.pvsystem.retrieve_sam('cecmod').SunPower_SPR_E20_327)
+    Ns_module = device.N_s
+
+    # Scale relevant module parameters to single cell values
+    device.name = 'Si_cell_large'
+    device.Date = '6/16/2018'
+    device.N_s = 1  # Single large cell
+    device.V_oc_ref = device.V_oc_ref / Ns_module
+    device.V_mp_ref = device.V_mp_ref / Ns_module
+    device.a_ref = device.a_ref / Ns_module
+    device.R_s = device.R_s / Ns_module
+    device.R_sh_ref = Ns_module * device.R_sh_ref
+
+    return device
+
+
+def make_gold_dataset():
+    """
+    Make a gold dataset of I-V curves that accurately solve single-diode model.
+
+    The set of I-V curves are generated from a range of PV devices: small c-Si
+    cell, large c-Si cell, c-Si "mini-module", c-Si module, and CdTe module.
+    Additionally, each device is ideal, nominal, or degraded in terms of its
+    series and parallel resistances. I-V curves for each device are generated
+    over a matrix of irradiance and temperature. The I-V curve data are
+    returned in a python dict that is in a lossless, machine-interoperable
+    format, which is the expected input to most of the other functions in this
+    python module. (Other functions can convert this dict to a human-readable
+    format.) An explict method of Bishop [1] is used to compute (V, I) pairs
+    mostly in the first quadrant, and interval analysis guarantees the
+    numerical computations up to a specified tolerance.
+
+    New gold data can be added to the returned dict. Each change should be
+    versioned in ``dataset_version``, following SemVer and maintaining
+    backwards compatibility whenever possible. Be sure to update all dependent
+    functions.
+
+    Parameters
+    ----------
+
+    Returns
+    -------
+    gold_dataset : dict
+        The gold data are returned in a dict with metadata (e.g., timestamp,
```

Comment:
Is it practical to outline the `gold_dataset` dict here (i.e., list and describe each key), the device dict, and a list member dict for a device?